### ISSUE TITLE
Compatibility with python 3.7?

### ISSUE THREAD

**Issue by hmaarrfk** [vpelletier/python-libaio#12]
WHen trying to import libaio with python 3.7, i get

```python
12:10 $ python -c "import libaio"
Traceback (most recent call last):
  File "/home/mark/miniconda3/envs/libaio/lib/python3.7/ctypes/__init__.py", line 97, in CFUNCTYPE
    return _c_functype_cache[(restype, argtypes, flags)]
KeyError: (None, (<class 'libaio.libaio.LP_io_context'>, <class 'libaio.libaio.iocb'>, <class 'ctypes.c_long'>, <class 'ctypes.c_long'>), 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/mark/miniconda3/envs/libaio/lib/python3.7/site-packages/libaio/__init__.py", line 29, in <module>
    from . import libaio
  File "/home/mark/miniconda3/envs/libaio/lib/python3.7/site-packages/libaio/libaio.py", line 151, in <module>
    io_callback_t = CFUNCTYPE(None, io_context_t, iocb, c_long, c_long)
  File "/home/mark/miniconda3/envs/libaio/lib/python3.7/ctypes/__init__.py", line 99, in CFUNCTYPE
    class CFunctionType(_CFuncPtr):
TypeError: item 2 in _argtypes_ passes a union by value, which is unsupported.

```

Any ideas?

**Comment by hmaarrfk** [comment]
```
12:10 $ conda list
# packages in environment at /home/mark/miniconda3/envs/libaio:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       0_gnu    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
ld_impl_linux-64          2.34                 h53a641e_0    conda-forge
libaio                    0.3.111              h516909a_1    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.2.0                h24d8f2e_2    conda-forge
libgomp                   9.2.0                h24d8f2e_2    conda-forge
libstdcxx-ng              9.2.0                hdf63c60_2    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
openssl                   1.1.1f               h516909a_0    conda-forge
pip                       20.0.2                     py_2    conda-forge
python                    3.7.6           h8356626_5_cpython    conda-forge
python-libaio             0.7                        py_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
setuptools                46.1.3           py37hc8dfbb8_0    conda-forge
sqlite                    3.30.1               hcee41ef_0    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.5                h516909a_0    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge

```

**Comment by vpelletier** [comment]
Thanks for this report.

The cause is double-silly of me:
- mis-translation of libaio.h into libaio.py: `iocb` is supposed to be passed by address and not by value
- and this declaration is not even used in this whole module, as in libaio.h it is only used for parameters of helper functions which end up calling `io_set_callback` (which casts it into a void pointer), and the python implementation of these functions does not declare parameter type... (EDIT: also, the whole feature of having a callback is implemented in a pure-python callchain in `__init__.py`, IIRC to avoid the callback being garbage-collected while the corresponding IO block was still running, especially on interpreter shutdown, without having to keep strong references to callbacks inside this module...)

d68e948e261c20969f455815c947faeeeb00fbe0 should fix this issue. Sadly, none of the python versions I have complain about this mistake, as apparently the corresponding check was since reverted upstream. Could you check that it fixes the issue, and that it does not complain about any other function declarations ?

**Comment by hmaarrfk** [comment]
The commit you suggested fixes thing.

I had done it internally but didn't want to impose a fix on you since it was pretty easy to find.

it was also silly that i wasn't using that functionaly, but i didn't want to break things further down the line. In case the library was.

**Comment by hmaarrfk** [comment]
A new release would be appreciated ;)

**Comment by vpelletier** [comment]
Thanks for confirming it works. I just released the fixed version as 0.8 .

**Comment by hmaarrfk** [comment]
much appreciated!
